msal/src/main/java/com/microsoft/identity/client/AuthenticationResult.java

```diff
@@ -32,29 +32,52 @@
 import java.util.List;
 import java.util.concurrent.TimeUnit;
 
+import static com.microsoft.identity.common.internal.authscheme.PopAuthenticationSchemeInternal.SCHEME_POP;
+
 /**
  * MSAL successful authentication result. When auth succeeds, token will be wrapped into the
  * {@link AuthenticationResult} and passed back through the {@link AuthenticationCallback}.
  */
 public final class AuthenticationResult implements IAuthenticationResult {
 
+    private static final String SCHEME_DELIMITER = " ";
+
     private final String mTenantId;
     private final AccessTokenRecord mAccessToken;
     private final IAccount mAccount;
+    private final String mAuthorizationHeader;
 
-    public AuthenticationResult(@NonNull final List<ICacheRecord> cacheRecords) {
+    AuthenticationResult(@NonNull final List<ICacheRecord> cacheRecords,
+                         @NonNull final String authHeader) {
         final ICacheRecord mostRecentlyAuthorized = cacheRecords.get(0);
         mAccessToken = mostRecentlyAuthorized.getAccessToken();
         mTenantId = mostRecentlyAuthorized.getAccount().getRealm();
         mAccount = AccountAdapter.adapt(cacheRecords).get(0);
+        mAuthorizationHeader = authHeader;
     }
 
     @Override
     @NonNull
     public String getAccessToken() {
+        if (mAuthorizationHeader.startsWith(SCHEME_POP + SCHEME_DELIMITER)) {
```

Comment:
Hah! So this was originally implemented this way, but unfortunately doesn't work for broker scenarios.... (which I realized late and had to refactor 😅)

The `DevicePoPManager` singleton uses different keys depending on whether or not we're in the broker or a client app using MSAL, because of this we can't late-bind the signing because MSAL apps won't have access to the same signing keys used by the broker.

AKA

We can't sign the result once we get it back from broker, it instead needs to be signed _inside_ the broker and passed up/back

Reply:
Grab me or post more feedback if you have more to add here - in the meantime, I'll mark resolved and keep chugging